deltaspike/core/impl/src/main/java/org/apache/deltaspike/core/impl/future/DefaultFutureableStrategy.java

```diff
@@ -202,6 +203,13 @@ public Object call() throws Exception
         };
 
         final ExecutorService pool = getOrCreatePool(ic);
+        
+        if (void.class.isAssignableFrom(returnType))
```

Comment:
think void is enough, using Void is a workaround IMO and not a clean programming model we should promote. If we get some feedbacks justifying this we can add it later.

Alternative is to add a Handler depending the returned type and a kind of SPI but looks overkilled today.